nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-pubsub-processors/src/main/java/org/apache/nifi/processors/kafka/pubsub/PublishKafka.java

```diff
@@ -0,0 +1,350 @@
+            .defaultValue(RANDOM_PARTITIONING.getValue())
+            .required(false)
+            .build();
+    static final PropertyDescriptor COMPRESSION_CODEC = new PropertyDescriptor.Builder()
+            .name(ProducerConfig.COMPRESSION_TYPE_CONFIG)
+            .displayName("Compression Type")
+            .description("This parameter allows you to specify the compression codec for all data generated by this producer.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .allowableValues("none", "gzip", "snappy", "lz4")
+            .defaultValue("none")
+            .build();
+
+    static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("Any FlowFile that cannot be sent to Kafka will be routed to this Relationship")
+            .build();
+
+    static final List<PropertyDescriptor> descriptors;
+
+    static final Set<Relationship> relationships;
+
+
+    /*
+     * Will ensure that list of PropertyDescriptors is build only once, since
+     * all other lifecycle methods are invoked multiple times.
+     */
+    static {
+        List<PropertyDescriptor> _descriptors = new ArrayList<>();
+        _descriptors.addAll(sharedDescriptors);
+        _descriptors.add(DELIVERY_GUARANTEE);
+        _descriptors.add(KEY);
+        _descriptors.add(MESSAGE_DEMARCATOR);
+        _descriptors.add(META_WAIT_TIME);
+        _descriptors.add(PARTITION_CLASS);
+        _descriptors.add(COMPRESSION_CODEC);
+
+        descriptors = Collections.unmodifiableList(_descriptors);
+
+        Set<Relationship> _relationships = new HashSet<>();
+        _relationships.addAll(sharedRelationships);
+        _relationships.add(REL_FAILURE);
+        relationships = Collections.unmodifiableSet(_relationships);
+    }
+
+    /**
+     *
+     */
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    /**
+     *
+     */
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    /**
+     * Will rendezvous with Kafka if {@link ProcessSession} contains {@link FlowFile}
+     * producing a result {@link FlowFile}.
+     * <br>
+     * The result {@link FlowFile} that is successful is then transfered to {@link #REL_SUCCESS}
+     * <br>
+     * The result {@link FlowFile} that is failed is then transfered to {@link #REL_FAILURE}
+     *
+     */
+    @Override
+    protected void rendezvousWithKafka(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile != null) {
+            flowFile = this.doRendezvousWithKafka(flowFile, context, session);
+            if (!this.isFailedFlowFile(flowFile)) {
+                session.getProvenanceReporter().send(flowFile, context.getProperty(BOOTSTRAP_SERVERS)
```

Comment:
Good point, although I am removing ```kafka://`` (both consume and publish)`. It is not a real protocol and it can get confusing when multiple broker URLs are provided